dashboard/src/app/workspaces/workspace-details/environments/machine-config/edit-machine-name-dialog/edit-machine-name-dialog.controller.ts

```diff
@@ -17,30 +18,42 @@
  * @author Oleksii Kurinnyi
  */
 export class EditMachineNameDialogController {
+  private name: string;
+  private origName: string;
+  private machinesNames: string[];
+  private callbackController: WorkspaceMachineConfigController;
 
   /**
    * Default constructor that is using resource
    * @ngInject for Dependency injection
    */
-  constructor($mdDialog) {
+  constructor(private $mdDialog: ng.material.IDialogService) {
     this.$mdDialog = $mdDialog;
+
+    this.origName = this.name;
   }
 
-  isUnique(name) {
-    return this.machinesNames.indexOf(name) < 0;
+  isUnique(name: string): boolean {
+    if (name === this.origName) {
+      return true;
+    }
+    let nameRE = new RegExp('^' + name + '$', 'i');
+    return this.machinesNames.some((_name: string) => {
+      return nameRE.test(_name);
+    }) === false;
   }
```

Comment:
If it done because we show names in upercase - we can use old one check and use convert the string to uppercase or lowercase letters in bought cases(when we use machineNames and when we check input name name). But use machineNamesUppercase and  nameUppercase.  As for me it would be more clear.

WDYT?